core/src/main/java/com/opensymphony/xwork2/interceptor/AliasInterceptor.java

```diff
@@ -167,7 +182,13 @@ public void setAliasesKey(String aliasesKey) {
                 for (Object o : aliases.entrySet()) {
                     Map.Entry entry = (Map.Entry) o;
                     String name = entry.getKey().toString();
+                    if (isNotAcceptableExpression(name)) {
+                        continue;
+                    }
                     String alias = (String) entry.getValue();
+                    if (isNotAcceptableExpression(alias)) {
+                        continue;
+                    }
```

Comment:
Why did add this restrictions? User can define an alias `#{'application.myName':'myName'}` and this is a valid scenario, with your changes it won't work probably. Aso why do you want restrict internal expressions?

Reply:
I think it will work. On first evaluation (where I don't disturb), the result will be same as always, a java map with key `application.myName` and value `myName`. On second evaluation (here that I've added above codes), I validate the both key and value are valid and they're.

I think this answers the second part, right? I don't restrict the whole expression, I restrict key and value in the result Map individually to be valid because they're going to go for another evaluation.

Reply:
BTW I knew and understand that you probably mean that those are developer constant expressions and I shouldn't consider the first evaluation actually a real evaluation at this specific case for example, but unfortunately here in this interceptor I don't have an easy way to recognize if it's really evaluated and has parsed variables or not, in Component.java I had it, `!parsedResult.equals(findString(name))`. So unfortunately I can't for now easily distinguish between `#{'application.myName':'myName'}` and `#{ #parameters['name'] : #parameters['alias'] }` where in first one evaluation is just a constant to String but in second one it's reading from end user parameters!

Reply:
@lukaszlenart would you like resolve conversation if it addresses your concern? thanks again for your review!